storage-provider/server/src/pipeline/mod.rs

```diff
@@ -285,11 +322,17 @@ async fn add_piece(
     deal: DealProposal,
     deal_id: u64,
 ) -> Result<(), PipelineError> {
-    let mut sector = find_sector_for_piece(&state).await?;
-
     tracing::info!("Adding a piece...");
-
+    let mut sector = find_or_create_sector_for_piece(&state, deal.piece_size).await?;
+    // Check the height *after* getting the sector to get "the freshest" block
     let current_block = state.xt_client.height(true).await?;
+
+    let deal_expiration_distance = deal.end_block - current_block;
```

Comment:
When this can happen? I mean, deal has been already accepted, put into the pipeline, then it fails at this check. Storage Provider gets slashed god knows why. 

Shouldn't this be checked when we `publish/accept` deal and be sure it doesn't happen here?